src/Commands/ElasticsearchIndexCommand.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Rapidez\Core\Commands;
+
+use Carbon\Carbon;
+use Exception;
+use Rapidez\Core\Commands\InteractsWithElasticsearchCommand;
+
+abstract class ElasticsearchIndexCommand extends InteractsWithElasticsearchCommand
+{
+    /**
+     * Index all items in all stores
+     *
+     * @param  string $indexName
+     * @param  (callable(object): array)|array $data
+     * @param  (callable(object): array)|array $values
+     */
+    public function indexAll($indexName, $data, $values)
+    {
+        $stores = $this->getStores();
+        foreach ($stores as $store) {
+            $this->indexStore($store, $indexName, $data, $values);
+        }
+    }
+
+    /**
+     * Index all items in a specific stores
+     *
+     * @param  object $store
+     * @param  string $indexName
+     * @param  (callable(object): array)|array $data
+     * @param  (callable(object): array)|array $values
+     */
+    public function indexStore($store, $indexName, $data, $values)
+    {
+        $this->line('Indexing `'.$indexName.'` for store: '.$store->name);
+        try {
+            [$alias, $index] = $this->prepareIndexer($store, $indexName);
+
+            $currentData = is_callable($data)
+                ? $data(config()->get('rapidez.store_code'))
+                : $data;
```

Comment:
Laravel has a function to handle this for you already 🙂 
https://github.com/laravel/framework/blob/82875a158e8f130a9103e096419f8518161b34d3/src/Illuminate/Collections/helpers.php#L186
```suggestion
            $currentData = value($data, config()->get('rapidez.store_code'));
```